docs/src/model_setup/output_writers.md

```diff
@@ -106,6 +106,7 @@ provided that their `dimensions` are provided:
 
 ```jldoctest
 using Oceananigans
+using Oceananigans.Grids: xnodes_reshaped, ynodes_reshaped
```

Comment:
should these functions be called "reshaped_xnodes", etc (eg as one would say in english)? Why introduce new names rather than using `xnodes(args... ; reshaped=true)`?

Reply:
I can change the names, no problem. I chose these names because a user can type x, press tab on REPL, and "discover" these functions. But I don't have strong feeling about it.

I created these separate functions because the compiler can't dispatch on keyword arguments. And because of that I was running into problems specializing the `xnodes` functions properly for different grids (given that now we also have the `with_halos` flag).

It's possible to do it, but it would make the `xnodes` code significantly more complex. Since the reshaped version of these functions are only used in some tests and in one `NetCDFOutputWriter`'s example throughout the whole code, it didn't feel like it was worth complicating the main `xnode()` function for it.

In fact, I feel we could nuke these reshaped functions and just reshape arrays manually in the few places in the code we use them.

Thoughts?

Reply:
Note that reshaping is a user feature, not something that's needed internally as far as I know. Reshaping is needed for some plotting libraries (I think Plots and PyPlot), but not for Makie.

Reply:
As for the names, we've attempted to use the convention "write it as you would say it" to make it more natural to speak and communicate about the code. But we could prioritize REPL discovery instead. I think though its important to have consistent style, so we should make that style change across the code base, not just with one set of functions?

Reply:
Can you explain the difficulty in dispatching and point us to the place in the code this manifests?

Reply:
Right, I think we need the function definitions

```julia
function xnodes(grid; with_halos=false, reshape=false)
```

etc

Reply:
One can introduce a design f the form

```julia
# Below is the inner function, defined by specific grids
_xnodes(grid::ParticularGrid) = ...

# This is the outer user-facing function
function xnodes(grid; reshape=false, with_halos=false)
    x = _xnodes(grid)

    if with_halos
        # remove halos
    end

    if reshape
      # reshape x
    end

    return x
end
```

Reply:
> Note that reshaping is a user feature, not something that's needed internally as far as I know. Reshaping is needed for some plotting libraries (I think Plots and PyPlot), but not for Makie.

I just tested both of those and was able to plot heatmaps and lines without reshaping. Maybe it was an old version of those packages that needed it? I think since this is a user-feature for plotting, if none of the major plotting libraries require reshaping I vote we simply get rid of it. Are there any other major plotting packages that we could check?